### ISSUE TITLE
Bug Report: HSTS breaks long responses/FISH-7044

### ISSUE THREAD

**Issue by alexk201** [payara/Payara#6178]
### Brief Summary

Enabling HSTS breaks long responses in recent versions of Payara Server, including 5.2022.5 and 6.2023.1. 

### Expected Outcome

Enabling HSTS should add the respective headers to every HTTPS response but shouldn't effect the response in any other way.

### Current Outcome

Long responses (~2MB in my case) are being cut off by the server. The response is not closed correctly resulting in indefinite page loads and missing headers (e.g. content-length). The following stack trace appears in the server log:

```java
[#|2023-01-31T07:44:07.901+0000|WARNING|Payara 6.2023.3-SNAPSHOT|org.glassfish.grizzly.filterchain.DefaultFilterChain|_ThreadID=85;_ThreadName=http-thread-pool::http-listener-2(3);_TimeMillis=1675151047901;_LevelValue=900;|
  GRIZZLY0013: Exception during FilterChain execution
org.glassfish.grizzly.http.util.MimeHeaders$MaxHeaderCountExceededException: Illegal attempt to exceed the configured maximum number of headers: 100
        at org.glassfish.grizzly.http.util.MimeHeaders.createHeader(MimeHeaders.java:399)
        at org.glassfish.grizzly.http.util.MimeHeaders.addValue(MimeHeaders.java:428)
        at org.glassfish.grizzly.http.HttpHeader.addHeader(HttpHeader.java:656)
        at org.glassfish.grizzly.config.HSTSFilter.handleWrite(HSTSFilter.java:102)
        at org.glassfish.grizzly.filterchain.ExecutorResolver$8.execute(ExecutorResolver.java:81)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:246)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:178)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:118)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:96)
        at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:51)
        at org.glassfish.grizzly.filterchain.FilterChainContext.write(FilterChainContext.java:751)
        at org.glassfish.grizzly.filterchain.FilterChainContext.write(FilterChainContext.java:729)
        at org.glassfish.grizzly.http.io.OutputBuffer.flushBuffer(OutputBuffer.java:1004)
        at org.glassfish.grizzly.http.io.OutputBuffer.flushBinaryBuffers(OutputBuffer.java:980)
        at org.glassfish.grizzly.http.io.OutputBuffer.flushAllBuffers(OutputBuffer.java:952)
        at org.glassfish.grizzly.http.io.OutputBuffer.flush(OutputBuffer.java:696)
        at org.apache.catalina.connector.OutputBuffer.doFlush(OutputBuffer.java:275)
        at org.apache.catalina.connector.OutputBuffer.flush(OutputBuffer.java:259)
        at org.apache.catalina.connector.CoyoteOutputStream.flush(CoyoteOutputStream.java:173)
        at org.glassfish.jersey.servlet.internal.ResponseWriter$NonCloseableOutputStreamWrapper.flush(ResponseWriter.java:307)
        at org.glassfish.jersey.message.internal.CommittingOutputStream.flush(CommittingOutputStream.java:263)
        at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$UnCloseableOutputStream.flush(WriterInterceptorExecutor.java:281)
        at java.base/sun.nio.cs.StreamEncoder.implFlush(StreamEncoder.java:318)
        at java.base/sun.nio.cs.StreamEncoder.flush(StreamEncoder.java:153)
        at java.base/java.io.OutputStreamWriter.flush(OutputStreamWriter.java:251)
        at org.glassfish.jersey.message.internal.ReaderWriter.writeToAsString(ReaderWriter.java:170)
        at org.glassfish.jersey.message.internal.AbstractMessageReaderWriterProvider.writeToAsString(AbstractMessageReaderWriterProvider.java:107)
        at org.glassfish.jersey.message.internal.StringMessageProvider.writeTo(StringMessageProvider.java:76)
        at org.glassfish.jersey.message.internal.StringMessageProvider.writeTo(StringMessageProvider.java:36)
        at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.invokeWriteTo(WriterInterceptorExecutor.java:242)
        at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.aroundWriteTo(WriterInterceptorExecutor.java:227)
        at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:139)
        at org.glassfish.jersey.server.internal.JsonWithPaddingInterceptor.aroundWriteTo(JsonWithPaddingInterceptor.java:85)
        at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:139)
        at org.glassfish.jersey.server.internal.MappableExceptionWrapperInterceptor.aroundWriteTo(MappableExceptionWrapperInterceptor.java:61)
        at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:139)
        at org.glassfish.jersey.message.internal.MessageBodyFactory.writeTo(MessageBodyFactory.java:1116)
        at org.glassfish.jersey.server.ServerRuntime$Responder.writeResponse(ServerRuntime.java:670)
        at org.glassfish.jersey.server.ServerRuntime$Responder.processResponse(ServerRuntime.java:378)
        at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:368)
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:263)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:239)
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:697)
        at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:394)
        at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346)
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:357)
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:311)
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
        at org.apache.catalina.core.StandardWrapper.service(StandardWrapper.java:1569)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:259)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:167)
        at org.apache.catalina.core.StandardPipeline.doInvoke(StandardPipeline.java:757)
        at org.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:577)
        at com.sun.enterprise.web.WebPipeline.invoke(WebPipeline.java:99)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:158)
        at org.apache.catalina.connector.CoyoteAdapter.doService(CoyoteAdapter.java:372)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:239)
        at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:520)
        at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:217)
        at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:190)
        at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:535)
        at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:515)
        at java.base/java.lang.Thread.run(Thread.java:829)
|#]
```

### Reproducer

There is a testcase available at https://github.com/alexk201/payara-hsts, or you can follow these steps:
1. enable hsts on payara 6.2023.1 and restart the server
2. create and deploy an application containing a jax-rs webservice returning a long response (~2,000,000 characters to be safe)
3. when calling the webservice, the response will be cut off and not closed correctly, resulting in indefinite page loading and an incomplete body

### Operating System

Windows 11 Pro

### JDK Version

Zulu JDK 11.0.17+8-LTS

### Payara Distribution

Payara Server Full Profile, Payara Docker Image

**Comment by felixif** [comment]
Hi @alexk201,

Thank you for reporting this bug! I was able to reproduce it very easily using your reproducer! I have escalated this issue to our platform development team as `FISH-7044`. It may take a while to fix the issue. We always encourage you to contribute, so if you feel confident to do so, then please raise a PR and a member of our development team will gladly review. Otherwise this issue will be picked up in the future.

Thanks,
Felix

**Comment by alexk201** [comment]
This issue was fixed in versions 6.2023.9 (ce) and 6.6.0 (ee), thank you
